server/test/unit/createEventsArray.test.js

```diff
@@ -0,0 +1,387 @@
+"use strict";
+
+const { nanoid } = require("nanoid");
+const { createEventsArray } = require("../../utilities/createEventsArray");
+
+jest.mock("nanoid");
+const mockedNanoid = jest.mocked(nanoid);
+
+const commonFormFields = {
+  title: "test",
+  description: "test",
+  location: "test",
+  discordName: "test",
+  recurring: {
+    rate: "noRecurr",
+    days: [],
+  },
+};
+
+const commonEventFields = {
+  title: "test",
+  description: "test",
+  location: "test",
+  groupId: null,
+  rsvp: [],
+};
+
+const nonRecurring = [
+  {
+    description: "starts and ends on the same UTC day",
+    input: {
+      ...commonFormFields,
+      firstEventStart: Date.parse("2023-01-28T15:00:00.000+00:00"),
```

Comment:
I received typeerrors in all the tests in this sequence when I tried to run it! It looks like we're trying to feed the createEventsArray function an object with the properties "firstEventStart", "firstEventEnd", and "lastEventStart" when it's expecting to receive "initialDate", "finalDate", "startTime", and "endTime".